### PR TITLE
Queries for non-platform extensions will at least match the requested quarkus core version

### LINKED ISSUE #55: Non-platform extensions JSON should at least match the requested quarkus core version
Only show extensions which Quarkus core target is same or higher than the selected platform stream Quarkus core. Eg. if extension is targeting 2.0 it can be shown for 2.0+, if targeting 2.6 only show 2.6+

### PR DESCRIPTION
**Pull request by gastaldi**
- Bumps to Quarkus 2.5.2.Final
- Introduces a fix for the quarkus_core_version column being always '0.0.0'

Fixes #55

### CODE REVIEW

**Review by aloubyansky — APPROVED** [review]

**Comment by gastaldi** [comment]
There is only one drawback for this strategy: newer quarkus versions won't see older quarkiverse releases (eg. quarkus-prettytime won't be visible unless you release quarkus-prettytime again)

**Comment by gsmet** [comment]
Well, I think it doesn't fly. We are not releasing new versions for each Quarkus version. And there's no way we will do it.

**Comment by gastaldi** [comment]
@maxandersen any thoughts?

**Comment by gastaldi** [comment]
I'll provide the quarkus core version fix in a separate PR

**Comment by aloubyansky** [comment]
> There is only one drawback for this strategy: newer quarkus versions won't see older quarkiverse releases (eg. quarkus-prettytime won't be visible in 2.5.0.Final unless you release quarkus-prettytime using 2.5.0)

That was not the idea. If quarkus-prettytime was built with Quarkus 2.4.0.Final it should be included in the catalog based on 2.5.0.Final.

**Comment by aloubyansky** [comment]
However, the registry should not be always looking at the latest version of a given extension. E.g. if you release prettytime based on 2.6.0.Final, this prettytime release should not be listed in a catalog based on Quarkus 2.5.0.Final. However, for 2.5.0.Final a previous version of prettytime compatible with 2.5.0.Final should be included.

File: src/main/java/io/quarkus/registry/app/model/ExtensionRelease.java [hunk 1]
```diff
@@ -23,10 +23,12 @@
 @Entity
 @NamedQuery(name = "ExtensionRelease.findNonPlatformExtensions", query = "from ExtensionRelease ext " +
         "where ext.platforms is empty " +
+        "and ext.quarkusCoreVersionSortable >= :quarkusCore " +
```
Comment (gastaldi) [inline src/main/java/io/quarkus/registry/app/model/ExtensionRelease.java:26]: Looks like I inverted the condition here, it should have been `<=`?

**Comment by gastaldi** [comment]
> However, the registry should not be always looking at the latest version of a given extension. E.g. if you release prettytime based on 2.6.0.Final, this prettytime release should not be listed in a catalog based on Quarkus 2.5.0.Final. However, for 2.5.0.Final a previous version of prettytime compatible with 2.5.0.Final should be included.

Done

**Review by aloubyansky — APPROVED** [review]

**Comment by maxandersen** [comment]
Why would we need to release every quarkiverse extension?

The idea is a quarkiverse extension shouldn't show up unless there is a release that is at least compatible with a previous version. 

Meaning if the first release of quarkiverse-wonka targets quarkus 2.6 it won't show go in quarkus streams older than 2.6 but it will show up in 2.7 and higher. 

Currently it shows up for everything which is just not right.

**Pull request merged**